### PR TITLE
TEST/EXAMPLES: ucp_stream_recv_nb can return UCS_OK

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Fix ucp_client_server example
1) ucp_stream_recv_nb can return ucs_ok, it hasn't been handled before
2) added flag to ucp_stream_recv_nb to be sure that all the data was received

### CODE REVIEW

**Comment by Sergei-Lebedev** [comment]
@evgeny-leksikov please take a look

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by evgeny-leksikov** [comment]
ok to test

File: test/examples/ucp_client_server.c [hunk 1]
```diff
@@ -220,23 +225,24 @@ static int send_recv_stream(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server)
...
         if (UCS_PTR_IS_ERR(request)) {
             fprintf(stderr, "unable to receive UCX message (%s)\n",
                     ucs_status_string(UCS_PTR_STATUS(request)));
             ret = -1;
             goto out;
         } else {
             request_wait(ucp_worker, request);
-            printf("UCX data message was received\n");
         }
+        printf("UCX data message was received\n");
```
Comment (evgeny-leksikov) [inline test/examples/ucp_client_server.c]: why did you move the `printf`?
in general, request can be completed with error, lets return actual status from `request_wait` and print the message only if it's `UCS_OK`. you can obtain the status by `ucp_request_check_status` before freeing.

File: test/examples/ucp_client_server.c [hunk 2]
```diff
@@ -189,6 +189,11 @@ static void print_result(int is_server, char *recv_message)
  */
 static void request_wait(ucp_worker_h ucp_worker, test_req_t *request)
 {
+    /*  if operation was completed immediately */
+    if (request == NULL){
```
Comment (dmitrygladkov) [inline test/examples/ucp_client_server.c]: looks like there is missed space between `)` and `{`, should be
```
if (request == NULL) {
```

File: test/examples/ucp_client_server.c (hunk 1 above)
Comment (Sergei-Lebedev) [reply test/examples/ucp_client_server.c]: After I added check for UCS_OK send and recv become pretty similar, so I moved status check code into wait function. Is it better now?

File: test/examples/ucp_client_server.c (hunk 2 above)
Comment (Sergei-Lebedev) [reply test/examples/ucp_client_server.c]: fixed

File: test/examples/ucp_client_server.c [hunk 3]
```diff
@@ -187,20 +188,28 @@ static void print_result(int is_server, char *recv_message)
...
 {
+    ucs_status_t status;
+
+    if (UCS_PTR_IS_ERR(request)) {
+        return UCS_PTR_STATUS(request);
+    }
     /*  if operation was completed immediately */
-    if (request == NULL){
-        return;
+    else if (request == NULL) {
```
Comment (evgeny-leksikov) [inline test/examples/ucp_client_server.c]: No need else and pls put this branch first, before `UCS_PTR_IS_ERR`

File: test/examples/ucp_client_server.c [hunk 4]
```diff
@@ -214,40 +223,32 @@ static int send_recv_stream(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server)
...
-            goto out;
-        } else {
-            request_wait(ucp_worker, request);
-        }
-        printf("UCX data message was received\n");
+    }
+    status = request_wait(ucp_worker, request);
+    if (status != UCS_OK){
+        fprintf(stderr, "unable to %s UCX message (%s)\n",
+                is_server ? "recieve": "send",
```
Comment (evgeny-leksikov) [inline test/examples/ucp_client_server.c]: typo: recieve -> receive

**Comment by evgeny-leksikov** [comment]
bot:retest

**Comment by alinask** [comment]
@Sergei-Lebedev Was this issue found after a certain failure? If yes, can you share its details please?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10441/ for details (Mellanox internal link).

**Comment by Sergei-Lebedev** [comment]
@alinask  No, I just found that ucp_stream_recv_nb can return ucs_ok, while ucp_tag_recv_nb can return either error status or valid request pointer. Stream example doesn't illustrate that behaviour, so I decided to include check. Probably it's not easy to get failure in this example cause server starts first and message size is quite small.

**Comment by yosefe** [comment]
bot:pipe:retest

File: test/examples/ucp_client_server.c [hunk 5]
```diff
@@ -209,39 +224,32 @@ static int send_recv_stream(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server)
...
-            fprintf(stderr, "unable to send UCX message (%s)\n",
-                    ucs_status_string(UCS_PTR_STATUS(request)));
-            ret = -1;
-            goto out;
-        } else if (UCS_PTR_STATUS(request) != UCS_OK) {
-            request_wait(ucp_worker, request);
-        }
     } else {
         /* Server receives a message from the client using the stream API */
+
```
Comment (yosefe) [inline test/examples/ucp_client_server.c]: redundant space line

File: test/examples/ucp_client_server.c [hunk 6]
```diff
@@ -209,39 +224,32 @@ static int send_recv_stream(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server)
     test_req_t *request;
     size_t length;
     int ret = 0;
+    ucs_status_t status;
 
     if (!is_server) {
         /* Client sends a message to the server using the stream API */
+
```
Comment (yosefe) [inline test/examples/ucp_client_server.c]: redundant space line

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10479/ for details (Mellanox internal link).
